tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -152,13 +149,27 @@ impl Default for Format<Full, SystemTime> {
     }
 }
 
+#[cfg(feature = "json")]
+#[cfg_attr(docsrs, doc(cfg(feature = "json")))]
+impl<T> Format<Json, T> {
+    /// Use the full JSON format with event metadata flattened.
+    ///
+    /// See [`Json`].
+    #[cfg(feature = "json")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "json")))]
+    pub fn flattened_json(mut self, flatten_event: bool) -> Format<Json, T> {
```

Comment:
take it or leave it: I would probably call this `flatten_events` or something...since it's only implemented when the JSON formatter is in use, you're going to use it like
```rust
fmt::Layer::builder()
    // ...
    .json()
    .flattened_json()
     // ...
```
so it's already clear that the JSON formatting has been selected...